### PR TITLE
EOS-25472: Support Bundle - collect node health info - cpu/memory/disk space usage

### PR DESCRIPTION
**Pull request by satish-seagate**
Signed-off-by: Satish Darade <satish.darade@seagate.com>

# Problem Statement
- Enhancement: Support Bundle: collect node health info - CPU / memory / disk space usage

# Design
[root@storage-node1 common]# cat common/health/resources.json
{
    "resource_usage": {
        "cpu_usage": "28.9 %",
        "disk_usage": {
            "totalSpace": "21 GB",
            "usedSpace": "11 GB",
            "freeSpace": "8 GB",
            "diskUsedPercentage": "57.6 %"
        },
        "memory_usage": {
            "total": "7812 MB",
            "available": "4118 MB",
            "used_percent": "47.3 %",
            "used": "3270 MB",
            "free": "226 MB",
            "active": "4636 MB",
            "inactive": "2025 MB",
            "buffers": "494 MB",
            "cached": "3820 MB",
            "shared": "124 MB",
            "slab": "564 MB"
        }
    }
}
For more details please refer: https://jts.seagate.com/browse/EOS-25472

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: 
-  Confirm All CODACY errors are resolved [Y/N]: 

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [x] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [x] All the tests performed should be mentioned before Resolving a JIRA. 
- [x] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/utils/support_framework/support_bundle.py [hunk 1]
```diff
@@ -255,6 +275,69 @@ def get_status(bundle_id: str = None):
...
+                        "available": total_memory['available'],
+                        "used_percent": total_memory['percent'],
+                        "used": total_memory['used'],
+                        "free": total_memory['free'],
+                        "active": total_memory['active'],
+                        "inactive": total_memory['inactive'],
+                        "buffers": total_memory['buffers'],
+                        "cached": total_memory['cached'],
+                        "shared": total_memory['shared'],
+                        "slab": total_memory['slab']
```
Comment (cdeshmukh) [inline py-utils/src/utils/support_framework/support_bundle.py:336]: here for all these attributes, add comments about its meaning.

File: py-utils/src/utils/support_framework/support_bundle.py [hunk 2]
```diff
@@ -64,6 +67,40 @@ async def _generate_bundle(command):
         except FileExistsError:
             raise BundleError(errno.EINVAL, "Bundle ID already exists,"
                 "Please use Unique Bundle ID")
+
+        cortx_config_store = ConfigStore(config_url)
```
[the remark this answers is not in the document]
Comment (satish-seagate) [reply py-utils/src/utils/support_framework/support_bundle.py:71]: https://github.com/Seagate/cortx-utils/pull/600/files#r733862413

File: py-utils/src/utils/support_framework/support_bundle.py [hunk 3]
```diff
@@ -255,6 +275,69 @@ def get_status(bundle_id: str = None):
...
+        while i < index:
+            cpu_usage = f'{psutil.cpu_percent(interval=None, percpu=percpu)} %'
+            time.sleep(1)
+            i = i + 1
+        return cpu_usage
+
+    @staticmethod
+    def get_disk_overall_usage():
+        units_factor_GB = 1000000000
+        overall_usage = {
```
[the remark this answers is not in the document]
Comment (satish-seagate) [reply py-utils/src/utils/support_framework/support_bundle.py]: Done

File: py-utils/src/utils/support_framework/support_bundle.py [hunk 4]
```diff
@@ -105,6 +142,21 @@ async def _generate_bundle(command):
                 common_locations.add(const.CORTX_RELEASE_INFO.split('/')[1])
             else:
                 Log.warn(f'{const.CORTX_RELEASE_INFO} file not found.')
+            
+            # Adding node resources health into the support bundle.
+            health_target = os.path.join(bundle_path, 'common' + '/health')
+            os.makedirs(health_target, exist_ok = True)
+            with open(health_target + '/resources.json', 'w') as fp:
```
[the remark this answers is not in the document]
Comment (satish-seagate) [reply py-utils/src/utils/support_framework/support_bundle.py]: Done

**Pull request merged**